public/js/graph/react_node.js.jsx

```diff
@@ -204,59 +306,168 @@ var ReactNode = React.createClass({
     getInitialState: function() {
         var id = this.props.attributes['id'];
         var type = 'AND'; //Need to figure out whether it is a OR or AND
-        var status = 'inactive';
-        var parents = [];
-        var children = [];
-        var outEdges = [];
-        var inEdges = [];
-        
-        $('.path').map(function(key, element) {
-            if (id == element.getAttribute('data-target-node')){
-                parents.push(element.getAttribute('data-source-node'));
-                inEdges.push(element.id);
-            }
-            if (id == element.getAttribute('data-source-node')){
-                children.push(element.getAttribute('data-target-node'));
-                outEdges.push(element.id);
-            }
-        });
-        
-        if (parents.length == 0){
-            status = 'takeable';
-        }
-        
+        var status = this.props.parents.length == 0 ? 'takeable' : 'inactive';
+
         return {
             id: id,
-            parents: parents,
-            children: children,
-            outEdges: outEdges,
-            inEdges: inEdges,
             logicalType: type,
-            updated: false,
-            hybrid: this.props.hybrid,
-            status: status
+            status: status,
         };
     },
     
-    handleClick: function(event) {
-        //code here
+    isSelected: function () {
+        return this.state.status === 'active' || this.state.status === 'overridden';
     },
     
-    handleMouseEnter: function(event) {
-        var currentNode = this.refs[this.state.id];
-        //code here
+    arePrereqsSatisfied: function(svg) {
+        var sat = true;
+        function isAllTrue(element, index, array) {
+            //use ? operator
+            if (svg.refs['nodes'].refs[element]){
+                return svg.refs['nodes'].refs[element].isSelected();
+            }
+            if (svg.refs['bools'].refs[element]){
+                return svg.refs['bools'].refs[element].isSelected();
+            }
+        }
         
+        if (this.state.logicalType === 'AND') {
+            sat = this.props.parents.every(isAllTrue);     
+        } else if (this.state.logicalType === 'OR') {
+            sat = this.props.parents.some(isAllTrue);
+        }
+        return sat;
     },
     
-    handleMouseLeave: function(event) {
-        //code here
+    updateNode: function(svg, newState) {
+        var newNewState =''
+        if (this.arePrereqsSatisfied(svg)) {
+            if (this.isSelected() || this.props.hybrid) {
```

Comment:
@david-yz-liu I cleaned this up in my newest commit, `newState` won't be needed anymore.  I'm using `bind` in my callback function and it know the new updated state to work with.